test/example/stop

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: The RamenDR authors
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+
+import drenv
+
+if len(sys.argv) != 2:
+    print(f"Usage: {sys.argv[0]} cluster")
+    sys.exit(1)
+
+cluster = sys.argv[1]
+
+drenv.log_progress("Scaling down deployment")
+drenv.kubectl(
+    "scale",
+    "--replicas=0",
```

Comment:
(quesiton) The example deployment is scaled bacup up because start runs through the required start/test routines for the same when bringing up the cluster?

Reply:
Yes, when starting, the deployment remains scaled down, and running example/start scales it up.

Here is a test:

1. Creating and stopping a cluster
```
$ drenv start example.yaml; drenv stop example.yaml
2022-12-09 14:58:16,315 INFO    [example] Starting environment
2022-12-09 14:58:16,316 INFO    [ex1] Starting cluster
2022-12-09 14:58:17,318 INFO    [ex2] Starting cluster
2022-12-09 14:58:55,193 INFO    [ex1] Cluster started in 38.88 seconds
2022-12-09 14:58:55,193 INFO    [ex1/0] Running example/start
2022-12-09 14:58:55,460 INFO    [ex1/0] example/start completed in 0.27 seconds
2022-12-09 14:58:55,461 INFO    [ex1/0] Running example/test
2022-12-09 14:59:09,479 INFO    [ex1/0] example/test completed in 14.02 seconds
2022-12-09 14:59:10,741 INFO    [ex2] Cluster started in 53.42 seconds
2022-12-09 14:59:10,741 INFO    [ex2/0] Running example/start
2022-12-09 14:59:11,018 INFO    [ex2/0] example/start completed in 0.28 seconds
2022-12-09 14:59:11,018 INFO    [ex2/0] Running example/test
2022-12-09 14:59:26,261 INFO    [ex2/0] example/test completed in 15.24 seconds
2022-12-09 14:59:26,261 INFO    [example] Environment started in 69.95 seconds
2022-12-09 14:59:26,355 INFO    [example] Stopping environment
2022-12-09 14:59:26,779 INFO    [ex1/0] Running example/stop
2022-12-09 14:59:26,781 INFO    [ex2/0] Running example/stop
2022-12-09 14:59:30,282 INFO    [ex2/0] example/stop completed in 3.50 seconds
2022-12-09 14:59:30,282 INFO    [ex2] Stopping cluster
2022-12-09 14:59:31,441 INFO    [ex2] Cluster stopped in 1.16 seconds
2022-12-09 14:59:57,670 INFO    [ex1/0] example/stop completed in 30.89 seconds
2022-12-09 14:59:57,670 INFO    [ex1] Stopping cluster
2022-12-09 14:59:58,834 INFO    [ex1] Cluster stopped in 1.16 seconds
2022-12-09 14:59:58,835 INFO    [example] Environment stopped in 32.48 seconds
```

2. Starting cluster manually without running the scripts
```
$ minikube start -p ex1
😄  [ex1] minikube v1.28.0 on Fedora 36
    ▪ MINIKUBE_HOME=/data/minikube
✨  Using the kvm2 driver based on existing profile
👍  Starting control plane node ex1 in cluster ex1
🔄  Restarting existing kvm2 VM for "ex1" ...
📦  Preparing Kubernetes v1.25.3 on containerd 1.6.8 ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "ex1" cluster and "default" namespace by default
```

3. Checking deployment
```
$ kubectl get deploy -A --context ex1
NAMESPACE     NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
default       example-deployment   0/0     0            0           8m35s
kube-system   coredns              1/1     1            1           8m37s
```

4. Checking that waiting for deployments works (drenv.wait_for_deployments())
```
$ time kubectl wait deploy --all --all-namespaces --for=condition=available --context ex1 --timeout 300s
deployment.apps/example-deployment condition met
deployment.apps/coredns condition met

real	0m0.267s
user	0m0.080s
sys	0m0.021s
```
Interesting that the deployment is not available, but waiting for condition=available
works.

5. Running start scripts
```
$ example/start ex1; example/test ex1
* Deploying example
  deployment.apps/example-deployment unchanged
* Testing example deployment
  deployment "example-deployment" successfully rolled out
```

This means we cannot skip the start script to mitigate bad network. Do we have
a way to change deployment without persisting the change?

We can also try to pause the deployment, but I guess it will behave in the same
way and the deployment will remained paused when the cluster is started.

Reply:
> Interesting that the deployment is not available, but waiting for condition=available works.

That is mostly because the desired state and the actual state match (i.e need 0 replicas, and have 0 replicas)

> 
>     5. Running start scripts
> 
> 
> ```
> $ example/start ex1; example/test ex1
> * Deploying example
>   deployment.apps/example-deployment unchanged
> * Testing example deployment
>   deployment "example-deployment" successfully rolled out
> ```
> 
> This means we cannot skip the start script to mitigate bad network. Do we have a way to change deployment without persisting the change?

Hmm... not to my knowledge.

> 
> We can also try to pause the deployment, but I guess it will behave in the same way and the deployment will remained paused when the cluster is started.

True, what the desired state is set to will remain in the API server.

Reply:
I solved the issue in a much simpler way (#667), but I think the stop hook can be useful the regional dr and for future users of the tool. The current version includes only the stop hook commits.